core/components/ajaxupload/src/AjaxUpload.php

```diff
@@ -427,6 +428,23 @@ public function saveUploads($target, $clearQueue = false, $allowOverwrite = true
         }
         foreach ($this->session[$this->getOption('uid')] as $fileId => &$fileInfo) {
             if (file_exists($fileInfo['path'] . $fileInfo['uniqueName'])) {
+                if ($sanitizeFilename) {
+                    $pathinfo = pathinfo($fileInfo['originalName']);
+                    $fileName = $pathinfo['filename'];
+
+                    // Replace all spaces and special characters with -
+                    $fileName = strip_tags($fileName); // strip HTML
```

Comment:
I suggest to use $modx->filterPathSegment here.

Reply:
@Jako Thanks, I didn't know that existed. I implemented it, but it doesn't filter special characters or the problematic commas.. So I kept those replace rules in.

Reply:
Removed the separate string replacements for , and . characters. Those are stripped by the regex replace already. Guess I didn't test properly earlier..

Reply:
@Jako Hmm, the remaining regex replace also strips letters with accents, which is undesirable. Will change this still!